sourced/ml/algorithms/id_splitter/pipeline.py

```diff
@@ -0,0 +1,248 @@
+import argparse
+from datetime import datetime as dt
+import logging
+import pickle
+import random
+import os
+import warnings
+
+import numpy as np
+try:
+    from keras import backend as K
+    from keras.callbacks import CSVLogger, TensorBoard, ModelCheckpoint, LearningRateScheduler
+    import tensorflow as tf
+except ImportError as e:
+    warnings.warn("Tensorflow or/and Keras are not installed, dependent functionality is "
+                  "unavailable.")
+
+from sourced.ml.algorithms.id_splitter.features import prepare_features
+
+
+EPSILON = 10 ** -8
+DEFAULT_THRESHOLD = 0.5  # threshold that is used to binarize predictions of the model
+
+
+def set_random_seed(seed):
+    """
+    Fix random seed for reproducibility.
+    :param seed: seed value
+    """
+    np.random.seed(seed)
+    random.seed(seed)
+    tf.set_random_seed(seed)
+
+
+def to_binary(mat, threshold, inplace=True):
+    """
+    Helper function to binarize matrix
+    :param mat: matrix or array
+    :param threshold: if value >= threshold than it will be 1, else 0
+    :param inplace: whether modify mat inplace or not
+    :return: binarized matrix
+    """
+    mask = mat >= threshold
+    if inplace:
+        mat_ = mat
+    else:
+        mat_ = mat.copy()
+    mat_[mask] = 1
+    mat_[np.logical_not(mask)] = 0
+    return mat_
+
+
+def precision_np(y_true, y_pred, epsilon=EPSILON):
+    """
+    Precision metric.
+    :param y_true: ground truth labels - expect binary values
+    :param y_pred: predicted labels - expect binary values
+    :param epsilon: to avoid division by zero
+    :return: precision
+    """
+    true_positives = np.sum(y_true * y_pred)
+    predicted_positives = np.sum(y_pred)
+    return true_positives / (predicted_positives + epsilon)
+
+
+def recall_np(y_true, y_pred, epsilon=EPSILON):
+    """
+    Compute recall metric.
+    :param y_true: matrix with ground truth labels - expect binary values
+    :param y_pred: matrix with predicted labels - expect binary values
+    :param epsilon: added to denominator to avoid division by zero
+    :return: recall
+    """
+    true_positives = np.sum(y_true * y_pred)
+    possible_positives = np.sum(y_true)
+    return true_positives / (possible_positives + epsilon)
+
+
+def report(model, X, y, batch_size, threshold=DEFAULT_THRESHOLD, epsilon=EPSILON):
+    """
+    Prepare report for `model` on data `X` & `y`. It prints precision, recall, F1 score.
+    :param model: model to apply
+    :param X: features
+    :param y: labels (expected binary labels)
+    :param batch_size: batch size that will be used or prediction
+    :param threshold: threshold to binarize predictions
+    :param epsilon: added to denominator to avoid division by zero
+    """
+    log = logging.getLogger("quality-reporter")
```

Comment:
The log domain must always be the same as the name of the scope. Here - function name. You can additionally rename the function.

It is not the only place! Please fix all the logs.